lib/httpclient.rb

```diff
@@ -1240,7 +1240,7 @@ def do_get_block(req, proxy, conn, &block)
       conn.push(res)
       return res
     end
-    content = block ? nil : ''
+    content = block ? nil : String.new('')
```

Comment:
```suggestion
    content = block ? nil : ''.dup
```

Why not a simple `.dup`?

Reply:
Since these strings starts out as an empty string and the string is added later with the `<<` method, I was feeling like it seemed more appropriate to create empty string via String.new than duplicating an empty string. However, I do not have strong opinion on this so let me update them to use dup.